Flappy Bird Clone/Assets/Scripts/Obstacles/PointTriggerBehaviour.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using Player;
+using UnityEngine;
+
+namespace Obstacles
+{
+    public class PointTriggerBehaviour : MonoBehaviour
+    {
+        private BoxCollider2D _collider;
+        private bool _isPlayerOnTrigger;
+
+        private void Awake()
+        {
+            _collider = GetComponent<BoxCollider2D>();
+        }
+
+        // EVENTS
+        private void OnTriggerEnter2D(Collider2D other)
+        {
+            if (!_collider.isTrigger || !other.TryGetComponent(out PlayerBehaviour player))
+            {
+                return;
+            }
+            
+            _isPlayerOnTrigger = true;
+        }
+
+        private void OnTriggerExit2D(Collider2D other)
+        {
+            if (_isPlayerOnTrigger)
```

Comment:
It's a bit confusing. Why you never reset this variable?

Reply:
Here I forgot to reset them 😅